### ISSUE TITLE
Allow "scrolling" through results in the list. 

### ISSUE THREAD

**Issue by rockerBOO** [nvim-lua/telescope.nvim#42]
Right now you are limited to the results in the list. 

If I am on vertical I am only showing 8 results by default. 

I sometimes use the fuzzy to give me a short list to browse through. 

![](https://i.imgur.com/4i8UZIy.png)

## Solutions

Allow us to go up on the list beyond the presented list.

I think limiting the the total results would be fine. 100 total results available for instance. Unless we want to encourage fuzzy browsing which then no limit (high limit) might be good.

**Comment by clason** [comment]
You know, until now I didn't realize that you _couldn't_ scroll through the results list...

(I think I'd rather do lazy loading, i.e., if you scroll to the top-most result, retrigger the search with the current typed pattern and cut off at `2*results.height`.)

**Comment by tjdevries** [comment]
Yes, it's a good idea. It might require a bit of time to think about. I have implemented `lines_manager` to (potentially) be able to handle this kind of thing at some point in the future.

I think maybe first step would be:
1. make it so that we can have top->down or down->top results sorting. This will require me to find all the places I have ssumptions about lines and sorting spots.
2. then, after doing that, we should be able to do some tricks to just move past the visible area.

**Comment by tami5** [comment]
I'm starting to use loclist to get tags (from markdown into location) and use it as an outline. This feature is really useful

**Comment by mizlan** [comment]
> Yes, it's a good idea. It might require a bit of time to think about. I have implemented `lines_manager` to (potentially) be able to handle this kind of thing at some point in the future.

@tjdevries Where can I find `lines_manager`? Is it on a particular branch or fork?

**Comment by apazzolini** [comment]
Until this feature is built, are the full results accessible programmatically?

Specifically, I'd like to be able to trigger live_grep and send all results (including the ones outside the visible range) to the quickfix list.

**Comment by VVKot** [comment]
@apazzolini you can do exactly that with `<C-q>` https://github.com/nvim-telescope/telescope.nvim/blob/f92b9b1fae70d5fac681a29f0df64549c399f18f/lua/telescope/mappings.lua#L29

**Comment by apazzolini** [comment]
> actions.send_to_qflist + actions.open_qflist

Oh, I didn't realize that sent invisible results as well. Thanks!

**Comment by VVKot** [comment]
You're welcome. I remembered that I actually use `smart_send_to_qflist` instead, which sends all results if there is no selection - but only selected rows if there is. Check it out!

**Comment by FinallyFinancialFreedom** [comment]
yes, this feature is highly wanted, i leave a comment in order to be notified at the first time.

**Comment by VVKot** [comment]
@FinallyFinancialFreedom // whoever else who wants to subscribe to issues, there is a subscribe button on the right you can click without a need to comment:

<img width="375" alt="Screenshot 2021-06-06 at 05 34 09" src="https://user-images.githubusercontent.com/24480985/120912640-d49f6580-c688-11eb-82d2-3b2c2b415b88.png">

**Comment by tjdevries** [comment]
+1

**Comment by her** [comment]
@tjdevries add a bounty to it? :)

**Comment by tjdevries** [comment]
Haha, well it's on the todo list. I completed step one which was asyncify more of the Ui. This is one of the next steps.

**Comment by tjdevries** [comment]
I don't know what the easiest way to set up bounty or fundraiser is
